### ISSUE TITLE
drop IKEv1's split SA install

### ISSUE THREAD

**Issue by cagney** [libreswan/libreswan#1363]
On IKEv1, on the responder, the SA is installed in two steps:

- receive quick-i1
- install inbound SA
- send quick-r1

then:

- receive quick-i2
- install outbound SA

where as, with IKEv2:

- receive create-child-sa request
- install inbound & outbound
- send create-child-sa response

I suspect part of the logic behind it is that it allows the responder to install the inbound sa _before_ the initiator so that all is ready when the initiator starts sending data.

Is it really that important?

**Comment by letoams** [comment]
Yes, let’s not change that. The difference between v1 and v2 is an entire RTT.Sent using a virtual keyboard on a phoneOn Oct 31, 2023, at 12:42, cagney ***@***.***> wrote:﻿
On IKEv1, on the responder, the SA is installed in two steps:

receive quick-i1
install inbound SA
send quick-r1

then:

receive quick-i2
install outbound SA

where as, with IKEv2:

receive create-child-sa request
install inbound & outbound
send create-child-sa response

I suspect part of the logic behind it is that it allows the responder to install the inbound sa before the initiator so that all is ready when the initiator starts sending data.
Is it really that important?

**Comment by cagney** [comment]
The down side of keeping it is either a de-tuned IKEv2 SA install or maintaining an extra two code paths.

**Comment by bleve** [comment]
Right. But I agree with Paul, we shouldn't touch IKEv1 code path if we can avoid that.

**Comment by cagney** [comment]
> Yes, let’s not change that. The difference between v1 and v2 is an entire RTT.

IKEv1 takes 1.5 round trips, while IKEv2 takes 1, but that is an aside.

The difference is that IKEv1 always installs the inbound SA before the peer installs the matching outbound SA but IKEv2 doesn't make that guarantee - the responder installs it's outbound SA before the initiator installs the matching inbound SA.

IKEv1 felt this was important.  IKEv2 did not.

**Comment by cagney** [comment]
In the IKEv1 `alias-01` test a pexpect() is triggered because the routing is only half installed vis:
```
                /* is there a conflict */
                if (spd->wip.conflicting.owner.policy != NULL) {
                        /*
                         * Double check that it really does own the
                         * SPD.  After all it is about to trigger a
                         * reject.
                         */
                        struct connection *d = spd->wip.conflicting.owner.policy->connection;
                        if (!kernel_policy_installed(d)) {
                                connection_buf ocb;
                                llog_pexpect(logger, HERE,
                                             "conflicting %s policy for "PRI_CONNECTION" %s is not installed",
                                             enum_name_short(&routing_names, c->child.routing),
                                             pri_connection(d, &ocb),
                                             enum_name_short(&routing_names, d->child.routing));
                        }
                        connection_buf cb;
                        llog(RC_LOG_SERIOUS, logger,
                             "cannot install kernel policy -- it is in use for "PRI_CONNECTION,
                             pri_connection(spd->wip.conflicting.owner.policy->connection, &cb));
                        return false;
                }
```
ref #1412

**Comment by cagney** [comment]
Won't fix see 3a05581c1a361557b0f60874bd9299853e3a8d55
